unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs

```diff
@@ -88,6 +93,12 @@ void OnTriggerStay(Collider other)
             agent.isStopped = true;
         }
 
+        if (other.gameObject.tag == "SlowDown")
+        {
+            previouslyCollided = other.gameObject;
+            agent.speed = baseSpeed - (baseSpeed * slowedSpeedFactor);
```

Comment:
This is bringing to my attention how I've coded up modifiers such as boosted run speed. This seems a bit cleaner as it's done as a percentage. If it becomes a problem later on we'll worry about refactoring the calculations I did to be in terms of percentage if need be.